pkg/azure/api/api.go

```diff
@@ -193,9 +199,17 @@ func parseInterface(iface *network.Interface) (instanceID string, i *types.Azure
 	return
 }
 
+// deriveGatewayIP finds the default gateway for a given Azure subnet.
+// inspired by pkg/ipam/crd.go (as AWS, Azure reserves the first subnet IP for the gw).
+// Ref: https://docs.microsoft.com/en-us/azure/virtual-network/virtual-networks-faq
```

Comment:
Mind linking to `https://docs.microsoft.com/en-us/azure/virtual-network/virtual-networks-faq#are-there-any-restrictions-on-using-ip-addresses-within-these-subnets` ?